controller/common/manage_children.go

```diff
@@ -111,7 +111,7 @@ func deleteChildren(client *dynamicclientset.ResourceClient, parent *unstructure
 			// This observed object wasn't listed as desired.
 			glog.Infof("%v %v/%v: deleting %v %v", parent.GetKind(), parent.GetNamespace(), parent.GetName(), obj.GetKind(), obj.GetName())
 			uid := obj.GetUID()
-			err := client.Delete(name, &metav1.DeleteOptions{
+			err := client.Namespace(parent.GetNamespace()).Delete(name, &metav1.DeleteOptions{
```

Comment:
We can either set `Namespace` here, or earlier in the sync call tree via `dynClient.Kind(apiVersion, kind).Namespace(parent.GetNamespace())`. I prefer existing implementation because we may need to work with cluster-scoped `client` too.